pkg/hub/submarineraddonagent/agent.go

```diff
@@ -187,3 +208,18 @@ func (a *addOnAgent) permissionConfig(cluster *clusterv1.ManagedCluster, addon *
 
 	return operatorhelpers.NewMultiLineAggregate(errs)
 }
+
+func (a *addOnAgent) getCurrentPodImage() (string, error) {
+	podNamespce := os.Getenv("POD_NAMESPACE")
```

Comment:
i think we only needs to call this once before starting controller. And i think it is helpful if we have an agent-image flag. If it  is not set, we fall back to use this method to get image. if it fails,  we fatal the whole container.

Reply:
done